vlmeval/dataset/SGI_Bench_1_0/dry_experiment.py

```diff
@@ -0,0 +1,501 @@
+                    "main_code": prob["main_code"],
+                    "incomplete_main_code": prob["incomplete_main_code"],
+                    "incomplete_functions": prob["incomplete_functions"],
+                    "unit_test_0_data": prob["unit_test_0_data"],
+                    "unit_test_0_output": prob["unit_test_0_output"],
+                    "unit_test_1_data": prob["unit_test_1_data"],
+                    "unit_test_1_output": prob["unit_test_1_output"],
+                    "unit_test_2_data": prob["unit_test_2_data"],
+                    "unit_test_2_output": prob["unit_test_2_output"],
+                    "unit_test_3_data": prob["unit_test_3_data"],
+                    "unit_test_3_output": prob["unit_test_3_output"],
+                    "unit_test_4_data": prob["unit_test_4_data"],
+                    "unit_test_4_output": prob["unit_test_4_output"],
+                    "function_type": prob["function_type"],
+                    "runtime": prob["runtime"],
+                    "discipline": prob["discipline"],
+                    "direction": prob["direction"],
+                }
+            )
+            idx += 1
+        return pd.DataFrame(rows)
+
+    
+    def build_prompt(self, line):
+        if isinstance(line, int):
+            line = self.data.iloc[line]
+        question = line['question'] + """
+Output the completed function enclosed within <answer> and </answer> tags. 
+
+Example 1:
+<answer>
+def hello():
+    print("Hello")
+</answer>
+
+Example 2:
+<answer>
+def add(a, b):
+    return a+b
+
+def minus(a, b):
+    return a-b
+</answer>
+
+"""
+
+        msgs = [{'type': 'text', 'value': question}]
+        return msgs
+
+    def evaluate(self, eval_file, **judge_kwargs):
+        save_dir_last = 'sgi_code_logs'
+        global save_dir
+        work_dir = judge_kwargs.get('work_dir','./outputs')
+        save_dir = os.path.join(work_dir,save_dir_last)
+        tmp_data_dir_last = 'sgi_tmp_data'
+        global tmp_data_dir 
+        tmp_data_dir = os.path.join(work_dir,tmp_data_dir_last)
+        data = load(eval_file)
+        data = pd.DataFrame(data)
+        
+        ################################################################## 输入数据准备 ##################################################################
+        if not os.path.exists(os.path.join(save_dir, 'data_construction.json')):
+            os.makedirs(os.path.join(save_dir), exist_ok=True)
+            os.makedirs(os.path.join(tmp_data_dir), exist_ok=True)
+            os.makedirs(os.path.join(tmp_data_dir,"0206"), exist_ok=True)
+            os.makedirs(os.path.join(tmp_data_dir, "0200"), exist_ok=True)
+            os.makedirs(os.path.join(tmp_data_dir, "0236"), exist_ok=True)
+            
+            download_file("https://raw.githubusercontent.com/InternScience/SGI-Bench/main/evaluation/task_3_dry_experiment/data/SGI_DryExperiment_0206/t10k-images-idx3-ubyte.gz", tmp_data_dir+"/0206")
```

Comment:
Download the extra resource file to the subdir of `LMUDataRoot()`, instead of the evaluation work dir.